ironfish-cli/src/commands/chain/import.ts

```diff
@@ -0,0 +1,161 @@
+import os from 'os'
+import path from 'path'
+import tar from 'tar'
+import { v4 as uuid } from 'uuid'
+import { IronfishCommand } from '../../command'
+import { RemoteFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class ImportSnapshot extends IronfishCommand {
+  static hidden = false
+
+  static description = `Import chain snapshot`
+
+  static flags = {
+    ...RemoteFlags,
+    bucket: Flags.string({
+      char: 'b',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Bucket URL to download snapshot from',
+    }),
+    path: Flags.string({
+      char: 'p',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Path to snapshot file',
+    }),
+    confirm: Flags.boolean({
+      default: false,
+      description: 'confirm without asking',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(ImportSnapshot)
+
+    let snapshotPath
+    const tempDir = path.join(os.tmpdir(), uuid())
+    await fsAsync.mkdir(tempDir, { recursive: true })
+
+    if (flags.path) {
+      snapshotPath = flags.path
+    } else {
+      const bucket = (flags.bucket || DEFAULT_SNAPSHOT_BUCKET_URL || '').trim()
+      if (!bucket) {
+        this.log(`Cannot download snapshot without bucket URL`)
+      }
+
+      const client = await this.sdk.connectRpc()
+      const status = await client.getChainInfo()
+
+      const manifest = await axios
+        .get<{
+          checksum: string
+          file_name: string
+          file_size: number
+          timestamp: number
+          block_height: number
+        }>(`https://${bucket}/manifest.json`)
+        .then((r) => r.data)
+
+      if (!flags.confirm) {
+        this.log(
+          `This snapshot (${
+            manifest.file_name
+          }) contains the Iron Fish blockchain up to block ${
+            manifest.block_height
+          }. The size of the latest snapshot file is ${FileUtils.formatFileSize(
+            manifest.file_size,
+          )}`,
+        )
+
+        this.log(
+          `Current head sequence of your local chain: ${status.content.currentBlockIdentifier.index}`,
+        )
+
+        const confirm = await CliUx.ux.confirm('Do you wish to continue (Y/N)?')
+        if (!confirm) {
+          this.log('Snapshot download aborted.')
+          this.exit(0)
+        }
+      }
+
+      snapshotPath = path.join(tempDir, manifest.file_name)
+      const snapshotFile = await fsAsync.open(snapshotPath, 'w')
+      const bar = CliUx.ux.progress({
+        barCompleteChar: '\u2588',
+        barIncompleteChar: '\u2591',
+        format: 'Downloading snapshot: [{bar}] {value}% | ETA: {eta}s',
+      }) as ProgressBar
+
+      bar.start()
+
+      const hasher = crypto.createHash('sha256')
+      const writer = snapshotFile.createWriteStream()
+
+      await axios({
```

Comment:
This is using both `.then` and `await` -- I think it could be rewritten to use only await, like `const response = await axios({`